### PR TITLE
Jdbc daos

### PR DESCRIPTION
**Pull request by Vladyslav0711**

### CODE REVIEW

File: src/main/java/org/example/internetshop/dao/impl/ProductDaoImpl.java [hunk 1]
```diff
@@ -40,4 +40,14 @@ public Product update(Product product) {
...
+
+    @Override
+    public List<Product> getProductsByCart(Long cartId) {
+        return null;
+    }
+
+    @Override
+    public List<Product> getProductByOrder(Long orderId) {
+        return null;
+    }
```
Comment (dmytro-hryhoruk) [inline src/main/java/org/example/internetshop/dao/impl/ProductDaoImpl.java]: ??

File: src/main/java/org/example/internetshop/dao/jdbc/OrderDaoJdbcImpl.java [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+import org.example.internetshop.model.Order;
+import org.example.internetshop.model.Product;
+import org.example.internetshop.utill.ConnectionUtil;
+
+@Dao
+public class OrderDaoJdbcImpl implements OrderDao {
+    @Inject
+    ShoppingCartDao shoppingCartDao;
+    @Inject
+    ProductDao productDao;
```
Comment (dmytro-hryhoruk) [inline src/main/java/org/example/internetshop/dao/jdbc/OrderDaoJdbcImpl.java]: access modifiers

File: src/main/java/org/example/internetshop/dao/jdbc/ProductDaoJdbcImpl.java [hunk 3]
```diff
@@ -101,10 +107,55 @@ public boolean delete(Long id) {
...
+        String query = "SELECT id, product_name, price FROM shopping_carts_products scp"
+                + " JOIN products p ON scp.product_id = p.id"
+                + " WHERE scp.cart_id = ?;";
+        List<Product> products = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            setProducts(query, products, connection, cartId);
+            return products;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Getting products from cart failed", e);
+        }
```
Comment (dmytro-hryhoruk) [inline src/main/java/org/example/internetshop/dao/jdbc/ProductDaoJdbcImpl.java]: you need this method only in shoppingCartDao, so let's move it there, and you will get rid of reference between daos

File: src/main/java/org/example/internetshop/dao/jdbc/ProductDaoJdbcImpl.java [hunk 4]
```diff
@@ -101,10 +107,55 @@ public boolean delete(Long id) {
...
+    }
+
+    @Override
+    public List<Product> getProductByOrder(Long orderId) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/org/example/internetshop/dao/jdbc/ProductDaoJdbcImpl.java]: same about this one

File: src/main/java/org/example/internetshop/dao/jdbc/ShoppingCartJdbcImpl.java [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+        String query = "SELECT * FROM shopping_carts WHERE user_id=?";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, userId);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(getShoppingCartFromResultSet(resultSet));
+            } else {
+                return Optional.empty();
+            }
```
Comment (dmytro-hryhoruk) [inline src/main/java/org/example/internetshop/dao/jdbc/ShoppingCartJdbcImpl.java]: do we need else after return ?

File: src/main/java/org/example/internetshop/dao/jdbc/UserDaoJdbcImpl.java [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+        String query = "SELECT * FROM users WHERE login=?";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setString(1, login);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(getUserFromResultSet(resultSet));
+            } else {
+                return Optional.empty();
+            }
```
Comment (dmytro-hryhoruk) [inline src/main/java/org/example/internetshop/dao/jdbc/UserDaoJdbcImpl.java]: same

File: src/main/java/org/example/internetshop/dao/jdbc/UserDaoJdbcImpl.java [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+            statement.setString(2, user.getSurname());
+            statement.setString(3, user.getLogin());
+            statement.setString(4, user.getPassword());
+            statement.setLong(5, user.getId());
+            statement.executeUpdate();
+            query = "DELETE FROM users WHERE id=?;";
+            clearUserInfo(query, connection, user.getId());
+            setUserRoles(user);
+            ResultSet resultSet = statement.getResultSet();
+            return getUserFromResultSet(resultSet);
```
Comment (dmytro-hryhoruk) [inline src/main/java/org/example/internetshop/dao/jdbc/UserDaoJdbcImpl.java]: you already have this user, why should you create a new one?

File: src/main/java/org/example/internetshop/dao/jdbc/UserDaoJdbcImpl.java [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+    }
+
+    private void setUserRoles(User user) {
+        String query = "INSERT INTO users_roles (user_id, role_id)"
+                + "VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            for (Role role : user.getRoles()) {
+                statement.setLong(1, user.getId());
+                statement.setLong(2, role.getId());
```
Comment (dmytro-hryhoruk) [inline src/main/java/org/example/internetshop/dao/jdbc/UserDaoJdbcImpl.java:153]: where did you  get id on this stage?

Comment (Vladyslav0711) [reply src/main/java/org/example/internetshop/dao/jdbc/UserDaoJdbcImpl.java:153]: i setted them on role model. But, ok, i understood how to get it from db.

File: src/main/java/org/example/internetshop/dao/jdbc/OrderDaoJdbcImpl.java [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+        statement.setLong(1, orderId);
+        return statement.executeUpdate() != 0;
+    }
+
+    private void setProductsToOrder(Order order) {
+        String query = "INSERT INTO orders_products (order_id, product_id)"
+                + " VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            for (Product product : shoppingCartDao.get(order.getUserId()).get().getProducts()) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/org/example/internetshop/dao/jdbc/OrderDaoJdbcImpl.java]: you supposed to get Order with all the products

File: src/main/java/org/example/internetshop/dao/jdbc/UserDaoJdbcImpl.java [hunk 10]
```diff
@@ -0,0 +1,194 @@
...
+                + "WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setString(1, user.getName());
+            statement.setString(2, user.getSurname());
+            statement.setString(3, user.getLogin());
+            statement.setString(4, user.getPassword());
+            statement.setLong(5, user.getId());
+            statement.executeUpdate();
+            query = "DELETE FROM users WHERE id=?;";
```
Comment (dmytro-hryhoruk) [inline src/main/java/org/example/internetshop/dao/jdbc/UserDaoJdbcImpl.java]: is it okay?

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**